tests/draft2019-09/refRemote.json

```diff
@@ -129,6 +129,39 @@
             }
         ]
     },
+    {
+        "description": "base URI change - change folder in subschema with path from root",
+        "schema": {
+            "$id": "http://localhost:1234/scope_change_defs3.json",
+            "type" : "object",
+            "properties": {
+                "list": {"$ref": "#/$defs/baz/$defs/bar"}
+            },
+            "$defs": {
+                "baz": {
+                    "$id": "baseUriChangeFolderInSubschema2/",
+                    "$defs": {
+                        "bar": {
+                            "type": "array",
+                            "items": {"$ref": "folderInteger.json"}
+                        }
+                    }
+                }
+            }
+        },
+        "tests": [
+            {
+                "description": "number is valid",
+                "data": {"list": [1]},
+                "valid": true
+            },
+            {
+                "description": "string is invalid",
+                "data": {"list": ["a"]},
+                "valid": false
+            }
+        ]
+    },
```

Comment:
> a pointer that crosses schema boundaries doesn't make sense

Why not? The spec describes how any particular location can have multiple URIs that can address that location (although only one of them, which uses the base uri coming from the $id closest to it, is the canonical URI).

Reply:
Okay, I see: "An implementation MAY choose not to support addressing schemas by non-canonical URIs". So adding that test is not valid.

Reply:
FWIW I think that statement in the spec is bunk. I doubt that implementations would outright *not* support this -- rather, they would accidentally support this, but possibly get it wrong (as I did - the $ref worked fine but the new canonical uri after navigation was wrong, so the *second* $ref would get resolved wrong.)

Reply:
See https://json-schema.org/draft/2020-12/json-schema-core.html#rfc.section.9.2.1, especially,

> An implementation MAY choose not to support addressing schemas by non-canonical URIs.

I'm not a fan of how this concept is described in the spec, but you can generally think of pre-2019-09 `$id` to be analogous to `<base>` and post-2019-09 `$id` to be analogous to `<iframe>`.

Reply:
My implementation doesn't support this. When a schema is loaded, `$id`s are extracted and replaced with a reference. If there is a pointer that crosses an `$id`, there is an error because there is nothing there.

Reply:
I support it just fine (the document remains untouched (a json pointer offset from a known URI location will find the location deep in the document even if there are intermediary $ids) but I understand that others have chosen to go a different way. I'll move this test to my local test suite.